test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Enums/EnumsTest.cs

```diff
@@ -414,6 +415,50 @@ public async Task AddEntity()
             }
         }
 
+        [Fact]
+        public async Task PostToEnumCollection()
+        {
+            //Arrange
+            await ResetDatasource();
+            string requestUri = this.BaseAddress + "/convention/Employees/2/SkillSet?$format=application/json;odata.metadata=none";
+            //Get the count before the post
+            int count = 0;
+            using (HttpResponseMessage response = await this.Client.GetAsync(requestUri))
+            {
+                response.EnsureSuccessStatusCode();
+
+                var json = await response.Content.ReadAsObject<JObject>();
+                var result = json.GetValue("value") as JArray;
+                count = result.Count;
```

Comment:
>count [](start = 16, length = 5)

Same as here if you only want to get the count using "$count" segment.

Reply:
As above; this is validating the response from the POST, not a separate GET.

---
In reply to: [223519142](https://github.com/OData/WebApi/pull/1628#discussion_r223519142) [](ancestors = 223519142)